source/elements/oneDPL/source/parallel_api/iterators.rst

```diff
@@ -301,3 +361,84 @@ operation were applied to each of these iterators. The types ``T`` within the te
 
 ``make_zip_iterator`` constructs and returns an instance of ``zip_iterator``
 using the set of source iterators provided.
+
+Other Supported Iterators
++++++++++++++++++++++++++
+``std::reverse_iterator<IteratorT>`` is an ``AdaptingIteratorSource`` if ``IteratorT`` is an ``AdaptingIteratorSource``.
+``std::reverse_iterator<IteratorT>`` is an *indirectly device accessible iterator* if ``IteratorT`` is an *indirectly
+device accessible iterator*. The SYCL device-copyable requirement of ``std::reverse_iterator<IteratorT>`` for use in
+algorithms with a ``device_policy`` relies upon the trivial copyability of ``IteratorT`` and the specific implementation
+of ``std::reverse_iterator``. 
+
+Pointers are assumed to be USM shared or device memory pointers and are *indirectly device accessible*.
+Pointers are trivially copyable and therefore SYCL device-copyable.
+
+.. _iterators-device-accessible:
+
+Customization For User Defined Iterators
+++++++++++++++++++++++++++++++++++++++++
+oneDPL provides a mechanism to indicate whether custom iterators are *indirectly device accessible*.
+
+Applications may define a free function ``is_onedpl_indirectly_device_accessible(T)``, which accepts an argument of
+type ``T`` and returns a type with the base characteristic of ``std::true_type`` if ``T`` is *indirectly device accessible*,
+otherwise returns a type with the base characteristic of ``std::false_type``. The function must be discoverable by
+argument-dependent lookup (ADL). It may be provided as a forward declaration only, without defining a body.
+
+The return type of ``is_onedpl_indirectly_device_accessible`` is examined at compile time to determine if ``T`` is
+*indirectly device accessible*. The function overload to use must be selected with argument-dependent lookup.
+[*Note*: Therefore, according to the rules in the `C++ Standard`_, a derived type for which there is no
+function overload will match its most specific base type for which an overload exists. -- *end note*]
+
+Once ``is_onedpl_indirectly_device_accessible(T)`` is defined, the public trait
+``template<typename T> oneapi::dpl::is_indirectly_device_accessible[_v]`` will return the appropriate value. This public
+trait can also be used to define the return type of ``is_onedpl_indirectly_device_accessible(T)`` by applying it to any 
+source iterator component types. Refer to the example below.
+
+Example
+^^^^^^^
+
+The following example shows how to define a customization for `is_indirectly_device_accessible` trait for a simple
```

Comment:
```suggestion
The following example shows how to define a customization for ``is_indirectly_device_accessible`` trait for a simple
```
At least in the GitHub preview, this is needed to get the correct formatting.

Reply:
thanks, taken.